File: questao_14/questao_14.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct {
	configuracoes map[string]string
}

var instancia *Configuracao
var mu sync.Mutex

func GetInstancia() *Configuracao {
	mu.Lock()
	defer mu.Unlock()

	if instancia == nil {
		instancia = &Configuracao{
			configuracoes: make(map[string]string),
		}
	}
	return instancia
}

func (c *Configuracao) SetConfig(chave, valor string) {
	c.configuracoes[chave] = valor
}

func (c *Configuracao) GetConfig(chave string) string {
	return c.configuracoes[chave]
}

func main() {
	config1 := GetInstancia()
	config1.SetConfig("tema", "escuro")

	config2 := GetInstancia()
	fmt.Println(config2.GetConfig("tema"))

	fmt.Println(config1 == config2)
}

File: questao_14/questao_14.java
public class questao_14 {
    private static questao_14 instancia;

    private questao_14() {
    }

    public static questao_14 getInstancia() {
        if (instancia == null) {
            instancia = new questao_14();
        }
        return instancia;
    }

    public void setConfig(String chave, String valor) {
    }

    public String getConfig(String chave) {
        return null;
    }

    public static void main(String[] args) {
        questao_14 config1 = questao_14.getInstancia();
        config1.setConfig("tema", "escuro");

        questao_14 config2 = questao_14.getInstancia();
        System.out.println(config2.getConfig("tema"));

        System.out.println(config1 == config2);
    }
}
